src/qt/researcher/researcherwizard.cpp

```diff
@@ -0,0 +1,76 @@
+#include "qt/forms/ui_researcherwizard.h"
+#include "qt/researcher/researchermodel.h"
+#include "qt/researcher/researcherwizard.h"
+
+#include "logging.h"
+
+namespace {
+//!
+//! \brief Alias for the "start over" custom wizard button.
+//!
+constexpr QWizard::WizardButton START_OVER_BUTTON = QWizard::CustomButton1;
+} // Anonymous namespace
+
+// -----------------------------------------------------------------------------
+// Class: ResearcherWizard
+// -----------------------------------------------------------------------------
+
+ResearcherWizard::ResearcherWizard(
+    QWidget *parent,
+    ResearcherModel* researcher_model,
+    WalletModel* wallet_model)
+    : QWizard(parent)
+    , ui(new Ui::ResearcherWizard)
+    , m_researcher_model(researcher_model)
+{
+    ui->setupUi(this);
+    configureStartOverButton();
+
+    setAttribute(Qt::WA_DeleteOnClose, true);
+
+    ui->emailPage->setModel(researcher_model);
+    ui->projectsPage->setModel(researcher_model);
+    ui->beaconPage->setModel(researcher_model, wallet_model);
+    ui->authPage->setModel(researcher_model);
+    ui->summaryPage->setModel(researcher_model);
+
+    // This enables the beacon "renew" button on the summary page to switch the
+    // current page back to beacon page. Since the summary page cannot navigate
+    // back to a specific page directly, it emits a signal that we wire up here
+    // to change the page when requested:
+    //
+    connect(ui->summaryPage, SIGNAL(renewBeaconButtonClicked()),
+            this, SLOT(onRenewBeaconButtonClicked()));
+}
+
+ResearcherWizard::~ResearcherWizard()
+{
+    m_researcher_model->onWizardClose();
+
+    delete ui;
+}
+
+void ResearcherWizard::configureStartOverButton()
+{
+    setButtonText(START_OVER_BUTTON, tr("&Start Over"));
```

Comment:
Having this button always at the bottom as start over is a bit confusing if you are looking at beacon status and not setting up the beacon. Maybe have the button be renamed after you have a valid beacon to something like: Change Beacon

Reply:
But you might want to look at these screens without intending to change anything, so that is confusing too.

Reply:
I think that would be more clear than start over, however. If you weren't intending to change anything, I think it would at least be clear that you wouldn't need to press it. Could also try and go for something like  `Restart Beacon Creation` or `Redo Beacon` or something similar? It's hard to word it without the button sounding like it would renew the beacon.

Reply:
The protocol won't allow for changing a beacon unless it expires or the CPID changes (in these cases, it's a new beacon). With this in mind, starting over makes the most sense because the email address is the only parameter that will change the end result. In fact, the wallet will start from the beginning of the wizard after relaunching if BOINC changes the CPID out from under it since we have no way to track the previous CPID yet (that may be an enhancement for the future, but it's a rare occurrance). I left the ability to start over without changing anything in there mostly so that users can reassure themselves that they finished the process. 

The "Renew" button on the summary page will jump you back to the beacon page directly (without starting over) to resend the beacon after five months pass. Otherwise, you'll see this every time:

![image](https://user-images.githubusercontent.com/4282384/84948066-c095d900-b0b0-11ea-8cfe-0272323e3c01.png)

Reply:
I understand that it would be a new beacon when pressing the button; I was thinking of using change beacon to be changing parameters and changing to a new beacon. The last bit was me just saying that naming the button is difficult since saying something like `New Beacon` could be confused by a user as renewing.
Other potential alternatives: `Restart Configuration` or `Redo Configuration` or something like that could maybe work? Those would give users and idea of what would be started over if they were just checking their beacon status.

Reply:
I see what you're saying now. Naming stuff is hard indeed. I settled on "Start Over" because I couldn't think of a more accurate label and because we may add additional steps to this flow in the future. The generic label suffices for now, I think. We can tweak that later as needed.